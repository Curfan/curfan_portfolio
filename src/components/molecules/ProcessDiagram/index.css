.process-diagram {
	& > .step {
		display: flex;
		align-items: center;
		cursor: pointer;

		& > .label {
			margin-left: 25px;
			color: rgba(255, 255, 255, 0.12);
			transition: color 0.3s ease-in;
		}

		&.active {
			& > .label {
				color: white;
			}
		}
	}

	& > .process-line {
		padding: 16px;
	}
}
