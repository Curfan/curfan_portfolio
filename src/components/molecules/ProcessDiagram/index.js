import React, { useState, useEffect, Fragment } from 'react';
import classnames from 'classnames';

import Marker from 'components/atoms/Marker';
import ProcessLine from 'components/atoms/ProcessLine';

import styles from './index.css';

const ProcessDiagram = ({ className, process = [], onProcessChange = () => {} }) => {
	const [state, setState] = useState(0);

	useEffect(() => {
		onProcessChange(state);
	}, [state]);

	return (
		<div className={classnames(styles.processDiagram, className)}>
			{process.map((step, index) => (
				<Fragment key={step}>
					<div
						className={classnames(styles.step, { [styles.active]: state === index })}
						role="button"
						tabIndex="0"
						onClick={() => setState(index)}
						onKeyPress={() => {}}
					>
						<Marker active={state === index} />
						<div className={styles.label}>{step}</div>
					</div>
					{index !== process.length - 1 && <ProcessLine className={styles.processLine} />}
				</Fragment>
			))}
		</div>
	);
};

export default ProcessDiagram;
