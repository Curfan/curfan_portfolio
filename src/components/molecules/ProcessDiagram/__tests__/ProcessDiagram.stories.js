import React from 'react';
import { storiesOf } from '@storybook/react';

import ProcessDiagram from 'components/molecules/ProcessDiagram';

const stories = storiesOf('molecules/ProcessDiagram', module);

stories.add('__interactive', () => (
	<ProcessDiagram process={['Define', 'Design', 'Delivery', 'Validate']} />
));
