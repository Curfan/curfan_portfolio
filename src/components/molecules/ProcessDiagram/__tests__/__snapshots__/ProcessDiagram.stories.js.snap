// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="processDiagram"
  >
    <div
      class="step active"
      role="button"
      tabindex="0"
    >
      <div
        class="marker active"
      >
        <span />
      </div>
      <div
        class="label"
      >
        Define
      </div>
    </div>
    <div
      class="processLine processLine"
    >
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
    </div>
    <div
      class="step"
      role="button"
      tabindex="0"
    >
      <div
        class="marker"
      >
        <span />
      </div>
      <div
        class="label"
      >
        Design
      </div>
    </div>
    <div
      class="processLine processLine"
    >
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
    </div>
    <div
      class="step"
      role="button"
      tabindex="0"
    >
      <div
        class="marker"
      >
        <span />
      </div>
      <div
        class="label"
      >
        Delivery
      </div>
    </div>
    <div
      class="processLine processLine"
    >
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
      <span
        class="dash"
      />
    </div>
    <div
      class="step"
      role="button"
      tabindex="0"
    >
      <div
        class="marker"
      >
        <span />
      </div>
      <div
        class="label"
      >
        Validate
      </div>
    </div>
  </div>
</DocumentFragment>
`;
